main.tf

```diff
@@ -44,6 +44,10 @@ resource "aws_eip" "nat_instance_eips" {
   tags = merge(var.tags, {
     "Name" = "alternat-instance-${count.index}"
   })
+
+  lifecycle {
+    prevent_destroy = var.prevent_destroy_on_eips
+  }
```

Comment:
Does this actually work with a variable? There is a section about this in [the docs](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#literal-values-only):

> The lifecycle settings all affect how Terraform constructs and traverses the dependency graph. As a result, only literal values can be used because the processing happens too early for arbitrary expression evaluation.